battle-engine.js

```diff
@@ -2950,9 +2950,11 @@ Battle = (function () {
 			ignoreNegativeOffensive = true;
 			ignorePositiveDefensive = true;
 		}
-
-		var ignoreOffensive = !!(move.ignoreOffensive || (ignoreNegativeOffensive && atkBoosts < 0));
-		var ignoreDefensive = !!(move.ignoreDefensive || (ignorePositiveDefensive && defBoosts > 0));
+		// Check for abilities to see if defenses or offenses are negated (Unaware, Mold Breaker, etc.)
+		var targetAbilityIgnoreOffensive = (!target.ignore['Ability']) ? target.getAbility().ignoreOffensive : false;
```

Comment:
I wouldn't use `?:` unless a branch is necessary, personally.

I _would_ wrap everything to the right of an assignment in `()` if it had `&&` or `||`, though, just to make it more readable.